src/System.Text.Primitives/System/Text/InvariantParser_bool.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Text.Utf8;
+
+namespace System.Text
+{
+    public static partial class InvariantParser
+    {
+        #region Helper Methods
+        private static bool isTrue(byte[] utf8Text, int index)
+        {
+            if (utf8Text.Length - index < 4)
+                return false;
+
+            byte firstChar = utf8Text[index];
+            if (firstChar != 't' && firstChar != 'T')
+                return false;
+
+            byte secondChar = utf8Text[index + 1];
+            if (secondChar != 'r' && secondChar != 'R')
+                return false;
+
+            byte thirdChar = utf8Text[index + 2];
+            if (thirdChar != 'u' && thirdChar != 'U')
+                return false;
+
+            byte fourthChar = utf8Text[index + 3];
```

Comment:
I remember that JIT  removes bounds check when array is accessed in  the opposite order.
Like ```var fourthChar = array[3]; var firstChar = array[0];```. 
Interesting, will it work here?